lib/client.ml

```diff
@@ -95,6 +130,171 @@ let route client router msg =
   let msg = router msg in
   Inbox.push !client.inbox msg.category msg
 
+module Failure_detector = struct
+  (** Messages sent by the failure detector protocol *)
+  type message =
+    | Ping
+    | Acknowledge
+    | PingRequest of Address.t
+  [@@deriving bin_io]
+
+  (** Initializes the failure detection component
+  with a default state and given config *)
+  let make config =
+    { config; acknowledges = Base.Hashtbl.Poly.create (); sequence_number = 0 }
```

Comment:
How come we don't use standard library hashtable again?

Reply:
```ocaml
  let make config =
    {
      config;
      acknowledges =
        Base.Hashtbl.create ~growth_allowed:true ~size:0 (module Base.Int);
      sequence_number = 0;
    }
```
We use standard library: Int => see https://ocaml.janestreet.com/ocaml-core/v0.12/doc/base/Base/Hashtbl/index.html#val-create

If I misunderstood (which is of course possible, I know me 😇 ) we should rework the one within `peer.ml`

Reply:
I meant why don't we use `Hashtbl` instead of `Base.Hashtbl`, but it lacks a lot of convenient functions we use right now. Maybe it's fine like this. We can address in another issue (maybe #20 :sweat_smile:)

Reply:
I went for `Base.Hashtbl` on first without strong opinion on which one we should use 😅
Let's address the rework in #20